### PR TITLE
HIPP-103: Api Policy Conditions

### PR DESCRIPTION
**Pull request by llmikeyj**

### CODE REVIEW

File: conf/app.routes [hunk 1]
```diff
@@ -82,6 +82,10 @@ POST        /apis/add-an-api/change-select-application                         c
 GET         /apis                                                              controllers.HipApisController.onPageLoad()
 POST        /apis                                                              controllers.HipApisController.onSubmit()
 
+GET        /apis/add-an-api/apiPolicyConditionsDeclarationPage                 controllers.ApiPolicyConditionsDeclarationPageController.onPageLoad(mode: Mode = NormalMode)
+POST       /apis/add-an-api/apiPolicyConditionsDeclarationPage                 controllers.ApiPolicyConditionsDeclarationPageController.onSubmit(mode: Mode = NormalMode)
+GET        /apis/add-an-api/changeAPIPolicyConditionsDeclarationPage           controllers.ApiPolicyConditionsDeclarationPageController.onPageLoad(mode: Mode = CheckMode)
+POST       /apis/add-an-api/changeAPIPolicyConditionsDeclarationPage           controllers.ApiPolicyConditionsDeclarationPageController.onSubmit(mode: Mode = CheckMode)
```
Comment (PaulCDurham) [inline conf/app.routes:88]: URL for normal mode: /api-hub/apis/add-an-api/review-policy
URL for change mode: /api-hub/apis/add-an-api/change-review-policy

The scaffold adds the URLs based on lower camel case class name. HMRC Digital policy is to use all lower case with words separated by hyphens.

File: app/controllers/APIPolicyConditionsDeclarationPageController.scala [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Request, Result}
+import repositories.AddAnApiSessionRepository
+import services.ApiHubService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import views.html.ApiPolicyConditionsDeclarationPageView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class ApiPolicyConditionsDeclarationPageController @Inject()(
```
Comment (PaulCDurham) [inline app/controllers/APIPolicyConditionsDeclarationPageController.scala:37]: Class and file name mismatch. Could you change file name to match class name?

File: test/controllers/ApiPolicyConditionsDeclarationPageControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    "must return OK and the correct view for a GET when the user has selected an application" in {
+      val apiDetail = sampleApiDetail()
+      val application = FakeApplication
+      val fixture = buildFixture(Some(buildUserAnswers(apiDetail)))
+
+      when(fixture.apiHubService.getApiDetail(ArgumentMatchers.eq(apiDetail.id))(any()))
+        .thenReturn(Future.successful(Some(apiDetail)))
+
+      when(fixture.apiHubService.getUserApplications(ArgumentMatchers.eq(FakeUser.email.value), ArgumentMatchers.eq(true))(any()))
+        .thenReturn(Future.successful(Seq(application)))
```
Comment (PaulCDurham) [inline test/controllers/ApiPolicyConditionsDeclarationPageControllerSpec.scala:62]: Are there actually calls to getUserApplications?

File: test/controllers/ApiPolicyConditionsDeclarationPageControllerSpec.scala [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+      val apiDetail = sampleApiDetail()
+      val application = FakeApplication
+      val userAnswers = buildUserAnswers(apiDetail).set(AddAnApiSelectApplicationPage, application.id).toOption.value
+      val fixture = buildFixture(Some(userAnswers))
+
+      when(fixture.apiHubService.getApiDetail(ArgumentMatchers.eq(apiDetail.id))(any()))
+        .thenReturn(Future.successful(Some(apiDetail)))
+
+      when(fixture.apiHubService.getUserApplications(ArgumentMatchers.eq(FakeUser.email.value), ArgumentMatchers.eq(true))(any()))
+        .thenReturn(Future.successful(Seq(application)))
```
Comment (PaulCDurham) [inline test/controllers/ApiPolicyConditionsDeclarationPageControllerSpec.scala:85]: Are there actually calls to getUserApplications?

File: test/controllers/ApiPolicyConditionsDeclarationPageControllerSpec.scala [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+      running(fixture.application) {
+        val request = FakeRequest(POST, routes.ApiPolicyConditionsDeclarationPageController.onSubmit(NormalMode).url)
+          .withFormUrlEncodedBody(("value[0]", "accept"))
+        val result = route(fixture.application, request).value
+
+        status(result) mustBe SEE_OTHER
+        redirectLocation(result) mustBe Some(nextPage.url)
+      }
+    }
+
```
Comment (PaulCDurham) [inline test/controllers/ApiPolicyConditionsDeclarationPageControllerSpec.scala:116]: Could you add a test that the answer is set in the repo? The select application and select endpoints controller specs have examples. That's missing from the scaffold tests but I like to add it in.

**Review by PaulCDurham — CHANGES_REQUESTED** [review]

File: conf/app.routes (hunk 1 above)
Comment (llmikeyj) [reply conf/app.routes:88]: Ok

File: app/controllers/APIPolicyConditionsDeclarationPageController.scala (hunk 2 above)
Comment (llmikeyj) [reply app/controllers/APIPolicyConditionsDeclarationPageController.scala:37]: Ok

File: test/controllers/ApiPolicyConditionsDeclarationPageControllerSpec.scala (hunk 3 above)
Comment (llmikeyj) [reply test/controllers/ApiPolicyConditionsDeclarationPageControllerSpec.scala:62]: Ok

File: test/controllers/ApiPolicyConditionsDeclarationPageControllerSpec.scala (hunk 4 above)
Comment (llmikeyj) [reply test/controllers/ApiPolicyConditionsDeclarationPageControllerSpec.scala:85]: Ok

**Review by PaulCDurham — APPROVED** [review]

**Pull request merged**